### PR TITLE
Stop P/T Color of Morphs

### LINKED ISSUE #1126: P/T of face-down creature is coloured based on actual P/T of face-down card
I observed the following behaviour in a 1-player game:
- Put any creature card face-down on the battlefield (e.g. Ire Shaman)
- Set its P/T to the face-down card's P/T (e.g. 2/1)
  - Outcome: P/T numbers show in white.
- Set its P/T to something else (e.g. 2/2)
  - Outcome: P/T numbers show in red.

I haven't been had a chance to check whether opponents see the same thing in a 2-or-more-player game. If only the controller sees the difference in P/T colour, then it's a pretty minor bug. Otherwise, it has the potential to reveal hidden information to the opponent.

Git snapshot: 2de99475b03ccaa8b4eea397aa26e305d30fccdb

### PR DESCRIPTION
**Pull request by ZeldaZach**
This fixes #1126.

By including the `getFaceDown()` function from the `abstractcarditem.h` header, we can determine if the card that's about to have it's P/T changed is face up or face down. If the card is face down, we will write the text in white (the default color) no matter what the P/T of the creature/card really is. This is to prevent information from being leaked, such as, in the past, if you have a 2/2 that you morphed, the text would be white as the system would see it matching the database.

### CODE REVIEW

File: cockatrice/src/carditem.cpp [hunk 1]
```diff
@@ -16,6 +17,7 @@
 #include "settingscache.h"
 #include "tab_game.h"
 #include "pb/serverinfo_card.pb.h"
+#include "abstractcarditem.h"
```
Comment (poixen) [inline cockatrice/src/carditem.cpp]: Are these includes needed?

Comment (ZeldaZach) [inline cockatrice/src/carditem.cpp]: They were already there, but I can do some experiments and see if they're really needed.

Comment (poixen) [inline cockatrice/src/carditem.cpp]: Looks like you added QDebug and abstractcarditem

Comment (ZeldaZach) [inline cockatrice/src/carditem.cpp]: Abstractcarditem was needed to get the facedown status and QDebug was helping me to debug. No harm in having it in the file. Many files in the repo have it anyway.

Comment (poixen) [inline cockatrice/src/carditem.cpp]: CardItem **is** an AbstractCardItem, so will inherit its parents methods.

You can remove both imports and just have the `!getFaceDown()`

**Comment by ZeldaZach** [comment]
Removed the two unnecessary include statements per @poixen's review.

I believe this is ready to be merged.

**Comment by ZeldaZach** [comment]
Also, @Daenyth  with this update... it might be beneficial to advance the cockatrice service API number due to how this fixes something that was detrimental to game play and if your opponent is on an older version then they will have an advantage.

Can I get your opinion on that?

**Comment by poixen** [comment]
As it stands face down cards will *always* be orange. This is due to it being nameless and not being found in cards.xml. If there is no base p/t to compare to, it will be orange, it works the same way with star/star  cards, like Tarmgoyf.

There is no way to get an advantage over other players with how this currently is. We do not need to bump and API versions.

Also with this fix it means that ANY p/t will be white. I wanted white text to be the base for a card. Is a 999/999 facedown a base? I think it should be left as it is.

File: cockatrice/src/carditem.cpp [hunk 2]
```diff
@@ -110,7 +110,7 @@ void CardItem::paint(QPainter *painter, const QStyleOptionGraphicsItem *option,
         QStringList ptDbSplit = db->getCard(name)->getPowTough().split("/");
         QStringList ptSplit = pt.split("/");
         
-        if (ptDbSplit.at(0) != ptSplit.at(0) || ptDbSplit.at(1) != ptSplit.at(1))
+        if ((ptDbSplit.at(0) != ptSplit.at(0) || ptDbSplit.at(1) != ptSplit.at(1)) && !getFaceDown())
```
Comment (poixen) [inline cockatrice/src/carditem.cpp]: Change to: 
`if (facedown || ptDbSplit.at(0) != ptSplit.at(0) || ptDbSplit.at(1) != ptSplit.at(1))`

If the card is face down it should always be orange. Also performs this checks first so we dont waste cycles on comparisons if not needed.

**Comment by poixen** [comment]
Really this is a server side issue, the server should not be sending what the card really is if it is face down. I could "hack" my own build to make face down cards reveal some actual card data.

File: cockatrice/src/carditem.cpp (hunk 2 above)
Comment (ZeldaZach) [inline cockatrice/src/carditem.cpp]: I was thinking of making morphs white, but the orange is fine too. Changed :)

**Comment by poixen** [comment]
If you want you can also remove <QAction> as that import is not being used, though I'm bot even sure if this PR will be merged, seems like we want a server side fix.

**Comment by ctrlaltca** [comment]
Closing.

**Pull request closed without merging**